<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Differential gene expression | Portfolio</title>
  <meta name="description" content="Chapter 5 Differential gene expression | Portfolio" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Differential gene expression | Portfolio" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Differential gene expression | Portfolio" />
  
  
  

<meta name="author" content="alena savachenko" />


<meta name="date" content="2024-12-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="alignment.-ruwe-gegevens-inspecteren.html"/>
<link rel="next" href="creating-r-packages.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.10.4/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="assets/img/logo_toc.png"></a></li>
<li><a href="./">Data science for biology </a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction. CV</a></li>
<li class="chapter" data-level="2" data-path="genome-annotatie-ab-de-novo.html"><a href="genome-annotatie-ab-de-novo.html"><i class="fa fa-check"></i><b>2</b> Genome annotatie ab de novo.Inleiding</a></li>
<li class="chapter" data-level="3" data-path="samenvatting.html"><a href="samenvatting.html"><i class="fa fa-check"></i><b>3</b> Samenvatting</a>
<ul>
<li class="chapter" data-level="3.1" data-path="samenvatting.html"><a href="samenvatting.html#training-van-het-augustus-programma-voor-het-ontdekken-van-nieuwe-genmodellen-en-hun-patronen."><i class="fa fa-check"></i><b>3.1</b> 1.Training van het AUGUSTUS-programma voor het ontdekken van nieuwe genmodellen en hun patronen.</a></li>
<li class="chapter" data-level="3.2" data-path="samenvatting.html"><a href="samenvatting.html#de-training-van-het-augustus-programma-met-proteïne-van-langere-evolutionaire-afstand"><i class="fa fa-check"></i><b>3.2</b> 2.. De training van het AUGUSTUS-programma met proteïne van langere evolutionaire afstand</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html"><i class="fa fa-check"></i><b>4</b> Alignment. Ruwe gegevens inspecteren</a>
<ul>
<li class="chapter" data-level="4.1" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#genemarket.-model-opbouwen-protocol-1.-mrna-pijplijnne"><i class="fa fa-check"></i><b>4.1</b> GenemarkET. Model opbouwen (protocol 1). mRna pijplijnne</a></li>
<li class="chapter" data-level="4.2" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#deel-1.-model-opbouwen"><i class="fa fa-check"></i><b>4.2</b> Deel 1. Model opbouwen</a></li>
<li class="chapter" data-level="4.3" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#etrain-protocol7"><i class="fa fa-check"></i><b>4.3</b> Etrain (protocol7)</a></li>
<li class="chapter" data-level="4.4" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#prothints-en-de-eiwitpijplijn"><i class="fa fa-check"></i><b>4.4</b> ProtHints en de eiwitpijplijn</a></li>
<li class="chapter" data-level="4.5" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#prothints"><i class="fa fa-check"></i><b>4.5</b> ProtHints</a></li>
<li class="chapter" data-level="4.6" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#protocol-2.-het-creëren-van-genstructuren-voor-training-op-basis-van-eiwitten."><i class="fa fa-check"></i><b>4.6</b> Protocol 2. Het creëren van genstructuren voor training op basis van eiwitten.</a></li>
<li class="chapter" data-level="4.7" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#genomethreader"><i class="fa fa-check"></i><b>4.7</b> GenomeThreader</a></li>
<li class="chapter" data-level="4.8" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#protocol-6.verwijderen-van-redundant-genstructuren-protocol-6"><i class="fa fa-check"></i><b>4.8</b> Protocol 6.Verwijderen van Redundant Genstructuren (protocol 6)</a></li>
<li class="chapter" data-level="4.9" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#trainingsset-van-proteins.etrain"><i class="fa fa-check"></i><b>4.9</b> Trainingsset van Proteins.Etrain</a></li>
<li class="chapter" data-level="4.10" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#prediction-met-behulp-van-extrinsiek-bewijs-protocol-11-12"><i class="fa fa-check"></i><b>4.10</b> Prediction met behulp van extrinsiek bewijs (Protocol 11-12)</a></li>
<li class="chapter" data-level="4.11" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#identificatie-en-visualisatie"><i class="fa fa-check"></i><b>4.11</b> Identificatie en visualisatie</a></li>
<li class="chapter" data-level="4.12" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#gen-identificatie"><i class="fa fa-check"></i><b>4.12</b> Gen-identificatie</a></li>
<li class="chapter" data-level="4.13" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#visualisatie"><i class="fa fa-check"></i><b>4.13</b> Visualisatie</a></li>
<li class="chapter" data-level="4.14" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#genviz"><i class="fa fa-check"></i><b>4.14</b> GenViz</a></li>
<li class="chapter" data-level="4.15" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#jbrowse"><i class="fa fa-check"></i><b>4.15</b> JBrowse</a></li>
<li class="chapter" data-level="4.16" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#artemis"><i class="fa fa-check"></i><b>4.16</b> Artemis</a></li>
<li class="chapter" data-level="4.17" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#bijlage"><i class="fa fa-check"></i><b>4.17</b> Bijlage</a></li>
<li class="chapter" data-level="4.18" data-path="alignment.-ruwe-gegevens-inspecteren.html"><a href="alignment.-ruwe-gegevens-inspecteren.html#references"><i class="fa fa-check"></i><b>4.18</b> References</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="differential-gene-expression.html"><a href="differential-gene-expression.html"><i class="fa fa-check"></i><b>5</b> Differential gene expression</a></li>
<li class="chapter" data-level="6" data-path="creating-r-packages.html"><a href="creating-r-packages.html"><i class="fa fa-check"></i><b>6</b> Creating R packages</a></li>
<li class="chapter" data-level="7" data-path="genvisr.html"><a href="genvisr.html"><i class="fa fa-check"></i><b>7</b> GenVisR</a>
<ul>
<li class="chapter" data-level="7.1" data-path="genvisr.html"><a href="genvisr.html#introduction-to-dataset"><i class="fa fa-check"></i><b>7.1</b> Introduction to dataset</a></li>
<li class="chapter" data-level="7.2" data-path="genvisr.html"><a href="genvisr.html#implementation-genvizr"><i class="fa fa-check"></i><b>7.2</b> Implementation genVizR</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/alenagrrr3/portfolio_alena_savachenko" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Portfolio</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-gene-expression" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Differential gene expression<a href="differential-gene-expression.html#differential-gene-expression" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="differential-gene-expression.html#cb133-1" tabindex="-1"></a><span class="fu">library</span>(Rsubread)</span>
<span id="cb133-2"><a href="differential-gene-expression.html#cb133-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
<h3>
Introduction
</h3>
<p>Our goal is to see if the transcription factor ONECUT1 is a transcription factor(TF) that will be able to turn fibroblast cells into brain cells.
We will be doing this by researching the difference in gen expression between fibroblast treated with TF ONECUT1 and the control TF BCLXL.
The RNA-sequencing was performed with Illumina NextSeq 500 using HighOutput kit v2 for 75 cycles.
It was paired-end sequencing 2 × 43 bp.
The dataset was generated using a stranded-protocol.
The cells used were fibroblasts from a human female at the age of 29.
The fibroblasts lines were cultured in tissue culture flasks in Dulbecco’s modified Eagle’s medium containing 20% (vol/vol) fetal bovine serum, 1% (vol/vol) penicillin/streptomycin and 1% (vol/vol) sodium pyruvate.
Information from:(<a href="https://academic.oup.com/nar/article/47/11/5587/5485068?login=false" class="uri">https://academic.oup.com/nar/article/47/11/5587/5485068?login=false</a>)</p>
<h3>
Q score estimation
</h3>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="differential-gene-expression.html#cb134-1" tabindex="-1"></a><span class="fu">library</span>(png)</span>
<span id="cb134-2"><a href="differential-gene-expression.html#cb134-2" tabindex="-1"></a><span class="fu">library</span>(grid)</span>
<span id="cb134-3"><a href="differential-gene-expression.html#cb134-3" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;gridExtra&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     combine</code></pre>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="differential-gene-expression.html#cb137-1" tabindex="-1"></a>img2 <span class="ot">&lt;-</span>  <span class="fu">rasterGrob</span>(<span class="fu">as.raster</span>(<span class="fu">readPNG</span>(<span class="st">&quot;assets/img/SRR7866699_1_Per base sequence quality.png&quot;</span>)))</span>
<span id="cb137-2"><a href="differential-gene-expression.html#cb137-2" tabindex="-1"></a>img3 <span class="ot">&lt;-</span>  <span class="fu">rasterGrob</span>(<span class="fu">as.raster</span>(<span class="fu">readPNG</span>(<span class="st">&quot;assets/img/SRR7866699_2_Per base sequence quality.png&quot;</span>)))</span>
<span id="cb137-3"><a href="differential-gene-expression.html#cb137-3" tabindex="-1"></a></span>
<span id="cb137-4"><a href="differential-gene-expression.html#cb137-4" tabindex="-1"></a>top_annotation_Fig1 <span class="ot">&lt;-</span> <span class="fu">textGrob</span>(<span class="st">&quot;Overrepresented sequences from the forward read from SRR7866702&quot;</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">10</span>, <span class="at">font =</span> <span class="dv">8</span>))</span>
<span id="cb137-5"><a href="differential-gene-expression.html#cb137-5" tabindex="-1"></a>bottom_annotation_Fig1 <span class="ot">&lt;-</span> <span class="fu">textGrob</span>(<span class="st">&quot;Fig. 1: The Overrepresented sequences of SRR7866702_1 which is shown because it&#39;s the only one to have a known possible source.&quot;</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">7</span>, <span class="at">font =</span> <span class="dv">8</span>))</span>
<span id="cb137-6"><a href="differential-gene-expression.html#cb137-6" tabindex="-1"></a></span>
<span id="cb137-7"><a href="differential-gene-expression.html#cb137-7" tabindex="-1"></a>top_annotation_Fig2 <span class="ot">&lt;-</span> <span class="fu">textGrob</span>(<span class="st">&quot;Per base sequence quality from the forward read(left) and reverse read(right) from SRR66699&quot;</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">10</span>, <span class="at">font =</span> <span class="dv">8</span>))</span>
<span id="cb137-8"><a href="differential-gene-expression.html#cb137-8" tabindex="-1"></a>bottom_annotation_Fig2 <span class="ot">&lt;-</span> <span class="fu">textGrob</span>(<span class="st">&quot;Fig. 2: two graphs showing the per base sequence quality of SRR7866699_1(forward read) and SRR7866699_2(reverse read). </span></span>
<span id="cb137-9"><a href="differential-gene-expression.html#cb137-9" tabindex="-1"></a><span class="st">Both have an average Phred score &gt; 30 which means the reads are of high quality. </span></span>
<span id="cb137-10"><a href="differential-gene-expression.html#cb137-10" tabindex="-1"></a><span class="st">Of note is that not only in this figure but in all the fastqc&#39;s the forward reads have less variation in quality than the reverse reads.&quot;</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">7</span>, <span class="at">font =</span> <span class="dv">8</span>))</span>
<span id="cb137-11"><a href="differential-gene-expression.html#cb137-11" tabindex="-1"></a></span>
<span id="cb137-12"><a href="differential-gene-expression.html#cb137-12" tabindex="-1"></a><span class="co">#grid.arrange(img1, top=top_annotation_Fig1, bottom=bottom_annotation_Fig1)</span></span>
<span id="cb137-13"><a href="differential-gene-expression.html#cb137-13" tabindex="-1"></a><span class="fu">grid.arrange</span>(img2, img3, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">top=</span>top_annotation_Fig2, <span class="at">bottom=</span>bottom_annotation_Fig2)</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-101-1.png" width="100%" /></p>
<p></br>
</br></p>
<p>Overall the per base sequence average phred score is always above 30.
The per sequence phred score is mostly above 30.
In the forward read from the fibroblast treated with ONECUT1(run id:SRR7866702) in the Overrepresented sequences there is a hit suggesting that there is a possible contamination due to the TruSeq Adapter trimming sequence.
A noteworthy observation is that the from all the fastqc files the forward primers consistently have a higher average phred score, less variation in phred score and a bigger amount of Overrepresented sequences.</p>
<h3>
overview raw bam file
</h3>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="differential-gene-expression.html#cb138-1" tabindex="-1"></a><span class="fu">library</span>(Rsubread)</span>
<span id="cb138-2"><a href="differential-gene-expression.html#cb138-2" tabindex="-1"></a></span>
<span id="cb138-3"><a href="differential-gene-expression.html#cb138-3" tabindex="-1"></a>bam_dir<span class="ot">=</span><span class="st">&quot;rnaseq_onecut/bam&quot;</span></span>
<span id="cb138-4"><a href="differential-gene-expression.html#cb138-4" tabindex="-1"></a></span>
<span id="cb138-5"><a href="differential-gene-expression.html#cb138-5" tabindex="-1"></a>bam_files <span class="ot">&lt;-</span>  <span class="fu">list.files</span>(bam_dir, <span class="at">pattern =</span> <span class="st">&quot;(.70(0|1|2).*</span><span class="sc">\\</span><span class="st">.bam$)|(.699</span><span class="sc">\\</span><span class="st">.bam$)&quot;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb138-6"><a href="differential-gene-expression.html#cb138-6" tabindex="-1"></a>bam_files</span></code></pre></div>
<pre><code>## [1] &quot;rnaseq_onecut/bam/SRR7866699.bam&quot; &quot;rnaseq_onecut/bam/SRR7866700.bam&quot;
## [3] &quot;rnaseq_onecut/bam/SRR7866701.bam&quot; &quot;rnaseq_onecut/bam/SRR7866702.bam&quot;</code></pre>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="differential-gene-expression.html#cb140-1" tabindex="-1"></a>counts_dir <span class="ot">&lt;-</span> <span class="st">&quot;rnaseq_onecut/counts&quot;</span></span>
<span id="cb140-2"><a href="differential-gene-expression.html#cb140-2" tabindex="-1"></a></span>
<span id="cb140-3"><a href="differential-gene-expression.html#cb140-3" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> <span class="fu">featureCounts</span>(</span>
<span id="cb140-4"><a href="differential-gene-expression.html#cb140-4" tabindex="-1"></a>  </span>
<span id="cb140-5"><a href="differential-gene-expression.html#cb140-5" tabindex="-1"></a>  <span class="at">files =</span> bam_files,</span>
<span id="cb140-6"><a href="differential-gene-expression.html#cb140-6" tabindex="-1"></a>  <span class="at">annot.inbuilt =</span> <span class="st">&quot;hg38&quot;</span>,</span>
<span id="cb140-7"><a href="differential-gene-expression.html#cb140-7" tabindex="-1"></a>  <span class="at">useMetaFeatures =</span> <span class="cn">TRUE</span>,</span>
<span id="cb140-8"><a href="differential-gene-expression.html#cb140-8" tabindex="-1"></a>  <span class="at">strandSpecific =</span> <span class="dv">0</span>,</span>
<span id="cb140-9"><a href="differential-gene-expression.html#cb140-9" tabindex="-1"></a>  <span class="at">isPairedEnd =</span> <span class="cn">TRUE</span>, </span>
<span id="cb140-10"><a href="differential-gene-expression.html#cb140-10" tabindex="-1"></a>  <span class="at">countReadPairs =</span> <span class="cn">TRUE</span>, </span>
<span id="cb140-11"><a href="differential-gene-expression.html#cb140-11" tabindex="-1"></a>  <span class="at">nthreads =</span> <span class="dv">10</span></span>
<span id="cb140-12"><a href="differential-gene-expression.html#cb140-12" tabindex="-1"></a>)</span></code></pre></div>
<h3>
DESeqDataSetFromMatrix
</h3>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="differential-gene-expression.html#cb141-1" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb141-2"><a href="differential-gene-expression.html#cb141-2" tabindex="-1"></a><span class="fu">library</span>(GOstats)</span>
<span id="cb141-3"><a href="differential-gene-expression.html#cb141-3" tabindex="-1"></a><span class="fu">library</span>(GO.db)</span>
<span id="cb141-4"><a href="differential-gene-expression.html#cb141-4" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;org.Hs.eg.db&quot;</span>)</span>
<span id="cb141-5"><a href="differential-gene-expression.html#cb141-5" tabindex="-1"></a></span>
<span id="cb141-6"><a href="differential-gene-expression.html#cb141-6" tabindex="-1"></a></span>
<span id="cb141-7"><a href="differential-gene-expression.html#cb141-7" tabindex="-1"></a>alignment_stats_one <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file=</span><span class="st">&quot;rnaseq_onecut/bam/alignment_statistics.rds&quot;</span>)</span>
<span id="cb141-8"><a href="differential-gene-expression.html#cb141-8" tabindex="-1"></a></span>
<span id="cb141-9"><a href="differential-gene-expression.html#cb141-9" tabindex="-1"></a>rds_one <span class="ot">&lt;-</span>  <span class="fu">readRDS</span>(<span class="st">&quot;rnaseq_onecut/counts/read_counts_OC1.rds&quot;</span>)</span>
<span id="cb141-10"><a href="differential-gene-expression.html#cb141-10" tabindex="-1"></a></span>
<span id="cb141-11"><a href="differential-gene-expression.html#cb141-11" tabindex="-1"></a>count_matrix_one <span class="ot">&lt;-</span> rds_one<span class="sc">$</span>counts </span>
<span id="cb141-12"><a href="differential-gene-expression.html#cb141-12" tabindex="-1"></a></span>
<span id="cb141-13"><a href="differential-gene-expression.html#cb141-13" tabindex="-1"></a>metadata_one <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;rnaseq_onecut/onecut_sampledata_OC1.csv&quot;</span>)</span>
<span id="cb141-14"><a href="differential-gene-expression.html#cb141-14" tabindex="-1"></a></span>
<span id="cb141-15"><a href="differential-gene-expression.html#cb141-15" tabindex="-1"></a>metadata_one <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(metadata_one)</span>
<span id="cb141-16"><a href="differential-gene-expression.html#cb141-16" tabindex="-1"></a></span>
<span id="cb141-17"><a href="differential-gene-expression.html#cb141-17" tabindex="-1"></a><span class="fu">rownames</span>(metadata_one) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(metadata_one<span class="sc">$</span>Run, <span class="st">&quot;.bam&quot;</span>)</span>
<span id="cb141-18"><a href="differential-gene-expression.html#cb141-18" tabindex="-1"></a></span>
<span id="cb141-19"><a href="differential-gene-expression.html#cb141-19" tabindex="-1"></a><span class="fu">colnames</span>(count_matrix_one) <span class="sc">==</span> <span class="fu">rownames</span>(metadata_one)</span></code></pre></div>
<pre><code>## [1] TRUE TRUE TRUE TRUE</code></pre>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="differential-gene-expression.html#cb143-1" tabindex="-1"></a>metadata_one <span class="ot">&lt;-</span> metadata_one <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">cond_factor =</span> <span class="fu">str_replace_all</span>(Cell_type,<span class="fu">c</span>( <span class="st">&quot;Skin derived fibroblast overexpressing Bclxl&quot;</span><span class="ot">=</span><span class="st">&quot;Bclxl&quot;</span>, <span class="st">&quot;2 days after induction of OC1 in skin derived fibroblasts&quot;</span><span class="ot">=</span><span class="st">&quot;OC1&quot;</span>)))</span>
<span id="cb143-2"><a href="differential-gene-expression.html#cb143-2" tabindex="-1"></a></span>
<span id="cb143-3"><a href="differential-gene-expression.html#cb143-3" tabindex="-1"></a></span>
<span id="cb143-4"><a href="differential-gene-expression.html#cb143-4" tabindex="-1"></a>metadata_one<span class="sc">$</span>cond_factor <span class="ot">&lt;-</span> metadata_one<span class="sc">$</span>cond_factor <span class="sc">%&gt;%</span> <span class="fu">factor</span>(<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;Bclxl&quot;</span>, <span class="st">&quot;OC1&quot;</span>))</span>
<span id="cb143-5"><a href="differential-gene-expression.html#cb143-5" tabindex="-1"></a></span>
<span id="cb143-6"><a href="differential-gene-expression.html#cb143-6" tabindex="-1"></a>metadata_one<span class="sc">$</span>cond_factor</span></code></pre></div>
<pre><code>## [1] Bclxl Bclxl OC1   OC1  
## Levels: Bclxl OC1</code></pre>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="differential-gene-expression.html#cb145-1" tabindex="-1"></a>dds_one <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb145-2"><a href="differential-gene-expression.html#cb145-2" tabindex="-1"></a>  <span class="at">countData =</span> count_matrix_one,</span>
<span id="cb145-3"><a href="differential-gene-expression.html#cb145-3" tabindex="-1"></a>  <span class="at">colData =</span> metadata_one, </span>
<span id="cb145-4"><a href="differential-gene-expression.html#cb145-4" tabindex="-1"></a>  <span class="at">design =</span> <span class="sc">~</span> cond_factor</span>
<span id="cb145-5"><a href="differential-gene-expression.html#cb145-5" tabindex="-1"></a>)</span>
<span id="cb145-6"><a href="differential-gene-expression.html#cb145-6" tabindex="-1"></a>dds_normalized_one <span class="ot">&lt;-</span> <span class="fu">rlog</span>(dds_one)</span>
<span id="cb145-7"><a href="differential-gene-expression.html#cb145-7" tabindex="-1"></a>pca_one <span class="ot">&lt;-</span> dds_normalized_one  <span class="sc">%&gt;%</span> <span class="fu">assay</span>() <span class="sc">%&gt;%</span> <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">prcomp</span>()</span>
<span id="cb145-8"><a href="differential-gene-expression.html#cb145-8" tabindex="-1"></a>pca_one_summary <span class="ot">&lt;-</span> <span class="fu">summary</span>(pca_one)<span class="sc">$</span>importance</span>
<span id="cb145-9"><a href="differential-gene-expression.html#cb145-9" tabindex="-1"></a>pca_one_summary</span></code></pre></div>
<pre><code>##                             PC1      PC2      PC3          PC4
## Standard deviation     38.44093 9.561226 7.233422 2.326055e-13
## Proportion of Variance  0.91135 0.056380 0.032270 0.000000e+00
## Cumulative Proportion   0.91135 0.967730 1.000000 1.000000e+00</code></pre>
<h3>
PCA Plot
</h3>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="differential-gene-expression.html#cb147-1" tabindex="-1"></a>pca_one_plotting <span class="ot">&lt;-</span> <span class="fu">cbind</span>(metadata_one, pca_one <span class="sc">$</span>x)</span>
<span id="cb147-2"><a href="differential-gene-expression.html#cb147-2" tabindex="-1"></a>PC1_one_var <span class="ot">&lt;-</span> <span class="fu">round</span>(pca_one_summary[<span class="st">&quot;Proportion of Variance&quot;</span>, <span class="st">&quot;PC1&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits =</span> <span class="dv">1</span>)</span>
<span id="cb147-3"><a href="differential-gene-expression.html#cb147-3" tabindex="-1"></a>PC2_one_var <span class="ot">&lt;-</span> <span class="fu">round</span>(pca_one_summary[<span class="st">&quot;Proportion of Variance&quot;</span>, <span class="st">&quot;PC2&quot;</span>]<span class="sc">*</span><span class="dv">100</span>, <span class="at">digits =</span> <span class="dv">1</span>)</span>
<span id="cb147-4"><a href="differential-gene-expression.html#cb147-4" tabindex="-1"></a>pca_one_plotting <span class="ot">&lt;-</span> <span class="fu">cbind</span>(metadata_one, pca_one<span class="sc">$</span>x)</span>
<span id="cb147-5"><a href="differential-gene-expression.html#cb147-5" tabindex="-1"></a></span>
<span id="cb147-6"><a href="differential-gene-expression.html#cb147-6" tabindex="-1"></a>pca_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb147-7"><a href="differential-gene-expression.html#cb147-7" tabindex="-1"></a>  <span class="at">name=</span><span class="fu">c</span>(<span class="st">&quot;PC1&quot;</span>,<span class="st">&quot;PC2&quot;</span>,<span class="st">&quot;PC3&quot;</span>,<span class="st">&quot;PC4&quot;</span>) ,  </span>
<span id="cb147-8"><a href="differential-gene-expression.html#cb147-8" tabindex="-1"></a>  <span class="at">value=</span><span class="fu">c</span>(pca_one_summary[<span class="dv">2</span>,<span class="dv">1</span>]<span class="sc">*</span><span class="dv">100</span> ,pca_one_summary[<span class="dv">2</span>,<span class="dv">2</span>]<span class="sc">*</span><span class="dv">100</span> ,pca_one_summary[<span class="dv">2</span>,<span class="dv">3</span>]<span class="sc">*</span><span class="dv">100</span> ,pca_one_summary[<span class="dv">2</span>,<span class="dv">4</span>]<span class="sc">*</span><span class="dv">100</span> )</span>
<span id="cb147-9"><a href="differential-gene-expression.html#cb147-9" tabindex="-1"></a>  )</span>
<span id="cb147-10"><a href="differential-gene-expression.html#cb147-10" tabindex="-1"></a></span>
<span id="cb147-11"><a href="differential-gene-expression.html#cb147-11" tabindex="-1"></a><span class="co"># Barplot</span></span>
<span id="cb147-12"><a href="differential-gene-expression.html#cb147-12" tabindex="-1"></a><span class="fu">ggplot</span>(pca_data, <span class="fu">aes</span>(<span class="at">x=</span>name, <span class="at">y=</span>value, <span class="at">fill=</span>name)) <span class="sc">+</span> </span>
<span id="cb147-13"><a href="differential-gene-expression.html#cb147-13" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>)<span class="sc">+</span></span>
<span id="cb147-14"><a href="differential-gene-expression.html#cb147-14" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Variation percentage PC1 t/m PC4&quot;</span>)<span class="sc">+</span></span>
<span id="cb147-15"><a href="differential-gene-expression.html#cb147-15" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;PC1 t/m PC4&quot;</span>)<span class="sc">+</span></span>
<span id="cb147-16"><a href="differential-gene-expression.html#cb147-16" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Variation percentage, %&quot;</span>)<span class="sc">+</span></span>
<span id="cb147-17"><a href="differential-gene-expression.html#cb147-17" tabindex="-1"></a>        <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> value), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.2</span>)</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-105-1.png" width="672" /></p>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="differential-gene-expression.html#cb148-1" tabindex="-1"></a><span class="fu">ggplot</span>(pca_one_plotting) <span class="sc">+</span> </span>
<span id="cb148-2"><a href="differential-gene-expression.html#cb148-2" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>PC1, <span class="at">y=</span>PC2, <span class="at">color =</span> cond_factor, <span class="at">shape =</span> Cell_line), <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb148-3"><a href="differential-gene-expression.html#cb148-3" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;PCA for Bslxl study&quot;</span>) <span class="sc">+</span></span>
<span id="cb148-4"><a href="differential-gene-expression.html#cb148-4" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>, PC1_one_var, <span class="st">&quot;%)&quot;</span>)) <span class="sc">+</span></span>
<span id="cb148-5"><a href="differential-gene-expression.html#cb148-5" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>, PC2_one_var, <span class="st">&quot;%)&quot;</span>)) <span class="sc">+</span></span>
<span id="cb148-6"><a href="differential-gene-expression.html#cb148-6" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-106-1.png" width="672" /></p>
<p>PC1 and PC2 capture 91.1+5.6% of all variation,which together is 96.7%. PC1 contains the most important variations.
PC2 and PC3 are less important and PC4 has no significance.
Most significant difference between cells is if they are treated with BCLXL or ONECUT1 seen along the PC1 axis.
There is a slight difference between ONECUT1 cells along the PC2 axis, which refer to duplicate measurements. This observation is supported by a pheatmaps of activated genes, which also shows the light difference between duplo measurements in ONECUT1. The duplicate measurements of BCLXL are very close to each other along the PC2 axis and on the pheatmap.</p>
<h3>
Volcano Ploting
</h3>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="differential-gene-expression.html#cb149-1" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb149-2"><a href="differential-gene-expression.html#cb149-2" tabindex="-1"></a><span class="fu">library</span>(GO.db)</span>
<span id="cb149-3"><a href="differential-gene-expression.html#cb149-3" tabindex="-1"></a></span>
<span id="cb149-4"><a href="differential-gene-expression.html#cb149-4" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb149-5"><a href="differential-gene-expression.html#cb149-5" tabindex="-1"></a></span>
<span id="cb149-6"><a href="differential-gene-expression.html#cb149-6" tabindex="-1"></a>dge_one <span class="ot">&lt;-</span> (<span class="fu">DESeq</span>(dds_one))</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="differential-gene-expression.html#cb156-1" tabindex="-1"></a>dge_results_one <span class="ot">&lt;-</span> <span class="fu">results</span>(dge_one)</span>
<span id="cb156-2"><a href="differential-gene-expression.html#cb156-2" tabindex="-1"></a><span class="fu">summary</span>(dge_results_one)</span></code></pre></div>
<pre><code>## 
## out of 20818 with nonzero total read count
## adjusted p-value &lt; 0.1
## LFC &gt; 0 (up)       : 4376, 21%
## LFC &lt; 0 (down)     : 3883, 19%
## outliers [1]       : 0, 0%
## low counts [2]     : 3959, 19%
## (mean count &lt; 2)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="differential-gene-expression.html#cb158-1" tabindex="-1"></a>results_without_na_one <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(dge_results_one) <span class="sc">%&gt;%</span>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(padj))</span>
<span id="cb158-2"><a href="differential-gene-expression.html#cb158-2" tabindex="-1"></a>sign_genes <span class="ot">&lt;-</span> results_without_na_one[<span class="fu">which</span>(results_without_na_one<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.01</span>),]</span>
<span id="cb158-3"><a href="differential-gene-expression.html#cb158-3" tabindex="-1"></a>topGene <span class="ot">&lt;-</span> sign_genes[<span class="fu">which.max</span>(sign_genes<span class="sc">$</span>log2FoldChange),]</span>
<span id="cb158-4"><a href="differential-gene-expression.html#cb158-4" tabindex="-1"></a>topGene_name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(topGene)</span>
<span id="cb158-5"><a href="differential-gene-expression.html#cb158-5" tabindex="-1"></a></span>
<span id="cb158-6"><a href="differential-gene-expression.html#cb158-6" tabindex="-1"></a>dge_plotting <span class="ot">&lt;-</span> results_without_na_one <span class="sc">%&gt;%</span> </span>
<span id="cb158-7"><a href="differential-gene-expression.html#cb158-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">signif =</span> <span class="fu">if_else</span>(padj <span class="sc">&lt;</span> <span class="fl">0.01</span>, <span class="st">&quot;padj &lt; 0.01&quot;</span>, <span class="st">&quot;Not significant&quot;</span>))</span>
<span id="cb158-8"><a href="differential-gene-expression.html#cb158-8" tabindex="-1"></a></span>
<span id="cb158-9"><a href="differential-gene-expression.html#cb158-9" tabindex="-1"></a>dge_plotting<span class="sc">$</span>diffexpressed <span class="ot">&lt;-</span> <span class="st">&quot;NO&quot;</span></span>
<span id="cb158-10"><a href="differential-gene-expression.html#cb158-10" tabindex="-1"></a>dge_plotting<span class="sc">$</span>diffexpressed[dge_plotting<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> dge_plotting<span class="sc">$</span>pval <span class="sc">&lt;</span> <span class="fl">0.01</span>] <span class="ot">&lt;-</span> <span class="st">&quot;UP&quot;</span></span>
<span id="cb158-11"><a href="differential-gene-expression.html#cb158-11" tabindex="-1"></a>dge_plotting<span class="sc">$</span>diffexpressed[dge_plotting<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> dge_plotting<span class="sc">$</span>pval <span class="sc">&lt;</span> <span class="fl">0.01</span>] <span class="ot">&lt;-</span> <span class="st">&quot;DOWN&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="differential-gene-expression.html#cb159-1" tabindex="-1"></a>upreg <span class="ot">&lt;-</span> <span class="fu">sum</span>(dge_plotting<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">&quot;UP&quot;</span>)</span>
<span id="cb159-2"><a href="differential-gene-expression.html#cb159-2" tabindex="-1"></a>down <span class="ot">&lt;-</span>  <span class="fu">sum</span>(dge_plotting<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">&quot;DOWN&quot;</span>)</span>
<span id="cb159-3"><a href="differential-gene-expression.html#cb159-3" tabindex="-1"></a>notreg <span class="ot">&lt;-</span> <span class="fu">sum</span>(dge_plotting<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">&quot;NO&quot;</span>)</span>
<span id="cb159-4"><a href="differential-gene-expression.html#cb159-4" tabindex="-1"></a></span>
<span id="cb159-5"><a href="differential-gene-expression.html#cb159-5" tabindex="-1"></a>dge_plotting<span class="sc">$</span>delabel <span class="ot">&lt;-</span> <span class="fu">rownames</span>(dge_plotting)</span>
<span id="cb159-6"><a href="differential-gene-expression.html#cb159-6" tabindex="-1"></a></span>
<span id="cb159-7"><a href="differential-gene-expression.html#cb159-7" tabindex="-1"></a>dge_volcano <span class="ot">&lt;-</span>  dge_plotting <span class="sc">%&gt;%</span> </span>
<span id="cb159-8"><a href="differential-gene-expression.html#cb159-8" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">color =</span> diffexpressed)) <span class="sc">+</span></span>
<span id="cb159-9"><a href="differential-gene-expression.html#cb159-9" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb159-10"><a href="differential-gene-expression.html#cb159-10" tabindex="-1"></a>      <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.01</span>), <span class="at">col =</span> <span class="st">&quot;gray&quot;</span>, <span class="at">linetype =</span> <span class="st">&#39;dashed&#39;</span>)<span class="sc">+</span> </span>
<span id="cb159-11"><a href="differential-gene-expression.html#cb159-11" tabindex="-1"></a>      <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">col =</span> <span class="st">&quot;gray&quot;</span>, <span class="at">linetype =</span> <span class="st">&#39;dashed&#39;</span>) <span class="sc">+</span></span>
<span id="cb159-12"><a href="differential-gene-expression.html#cb159-12" tabindex="-1"></a>         <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">300</span>), <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb159-13"><a href="differential-gene-expression.html#cb159-13" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;log2 fold change&quot;</span>) <span class="sc">+</span></span>
<span id="cb159-14"><a href="differential-gene-expression.html#cb159-14" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;-log10 adjusted p-value&quot;</span>) <span class="sc">+</span> </span>
<span id="cb159-15"><a href="differential-gene-expression.html#cb159-15" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb159-16"><a href="differential-gene-expression.html#cb159-16" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(  <span class="st">&quot;#00AFBB&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;#bb0c00&quot;</span>), </span>
<span id="cb159-17"><a href="differential-gene-expression.html#cb159-17" tabindex="-1"></a>    </span>
<span id="cb159-18"><a href="differential-gene-expression.html#cb159-18" tabindex="-1"></a>    <span class="at">labels=</span><span class="fu">c</span>( <span class="fu">paste</span>( <span class="st">&quot;Downregulated,&quot;</span>, down) , <span class="fu">paste</span>(<span class="st">&quot;Not significant,&quot;</span>, notreg) ,<span class="fu">paste</span>(<span class="st">&quot;Upregulated,&quot;</span>, upreg)) ) </span>
<span id="cb159-19"><a href="differential-gene-expression.html#cb159-19" tabindex="-1"></a></span>
<span id="cb159-20"><a href="differential-gene-expression.html#cb159-20" tabindex="-1"></a>dge_volcano  </span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/dge2-1.png" width="672" /></p>
<h3>
Pheatmap ploting
</h3>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="differential-gene-expression.html#cb160-1" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb160-2"><a href="differential-gene-expression.html#cb160-2" tabindex="-1"></a></span>
<span id="cb160-3"><a href="differential-gene-expression.html#cb160-3" tabindex="-1"></a>upreg_rows <span class="ot">&lt;-</span> dge_plotting[dge_plotting<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">&quot;UP&quot;</span>,]</span>
<span id="cb160-4"><a href="differential-gene-expression.html#cb160-4" tabindex="-1"></a>up_reg_genes <span class="ot">&lt;-</span>  upreg_rows[<span class="fu">order</span>( upreg_rows<span class="sc">$</span>log2FoldChange,<span class="at">decreasing =</span> <span class="cn">TRUE</span>),]</span>
<span id="cb160-5"><a href="differential-gene-expression.html#cb160-5" tabindex="-1"></a></span>
<span id="cb160-6"><a href="differential-gene-expression.html#cb160-6" tabindex="-1"></a>top5_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(up_reg_genes[<span class="fu">order</span>(up_reg_genes<span class="sc">$</span>log2FoldChange <span class="sc">&amp;</span> up_reg_genes<span class="sc">$</span>padj)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>],])</span>
<span id="cb160-7"><a href="differential-gene-expression.html#cb160-7" tabindex="-1"></a>top5_genes</span></code></pre></div>
<pre><code>## [1] &quot;3175&quot;   &quot;10202&quot;  &quot;388228&quot; &quot;338440&quot; &quot;284422&quot;</code></pre>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="differential-gene-expression.html#cb162-1" tabindex="-1"></a>count_values_up <span class="ot">&lt;-</span> <span class="fu">assay</span>(dds_one)[top5_genes,]</span>
<span id="cb162-2"><a href="differential-gene-expression.html#cb162-2" tabindex="-1"></a></span>
<span id="cb162-3"><a href="differential-gene-expression.html#cb162-3" tabindex="-1"></a>down_rows <span class="ot">&lt;-</span> dge_plotting[dge_plotting<span class="sc">$</span>diffexpressed<span class="sc">==</span><span class="st">&quot;DOWN&quot;</span>,]</span>
<span id="cb162-4"><a href="differential-gene-expression.html#cb162-4" tabindex="-1"></a>down_reg_genes <span class="ot">&lt;-</span> down_rows[<span class="fu">order</span>( down_rows<span class="sc">$</span>log2FoldChange),]</span>
<span id="cb162-5"><a href="differential-gene-expression.html#cb162-5" tabindex="-1"></a></span>
<span id="cb162-6"><a href="differential-gene-expression.html#cb162-6" tabindex="-1"></a>top5_downereg_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(down_reg_genes[<span class="fu">order</span>(down_reg_genes<span class="sc">$</span>log2FoldChange <span class="sc">&amp;</span> down_reg_genes<span class="sc">$</span>padj)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>],])</span>
<span id="cb162-7"><a href="differential-gene-expression.html#cb162-7" tabindex="-1"></a>top5_downereg_genes</span></code></pre></div>
<pre><code>## [1] &quot;4973&quot;      &quot;1734&quot;      &quot;158584&quot;    &quot;102724434&quot; &quot;285313&quot;</code></pre>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="differential-gene-expression.html#cb164-1" tabindex="-1"></a>top_5_up_and_5_down_genes <span class="ot">&lt;-</span> <span class="fu">c</span>(top5_genes,top5_downereg_genes)</span>
<span id="cb164-2"><a href="differential-gene-expression.html#cb164-2" tabindex="-1"></a>top_5_up_and_5_down_genes</span></code></pre></div>
<pre><code>##  [1] &quot;3175&quot;      &quot;10202&quot;     &quot;388228&quot;    &quot;338440&quot;    &quot;284422&quot;    &quot;4973&quot;     
##  [7] &quot;1734&quot;      &quot;158584&quot;    &quot;102724434&quot; &quot;285313&quot;</code></pre>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="differential-gene-expression.html#cb166-1" tabindex="-1"></a>count_values_up_and_down <span class="ot">&lt;-</span> <span class="fu">assay</span>(dds_one)[top_5_up_and_5_down_genes,]</span>
<span id="cb166-2"><a href="differential-gene-expression.html#cb166-2" tabindex="-1"></a></span>
<span id="cb166-3"><a href="differential-gene-expression.html#cb166-3" tabindex="-1"></a><span class="fu">colnames</span>(count_values_up_and_down) <span class="ot">&lt;-</span> <span class="fu">colData</span>(dds_one)<span class="sc">$</span>cond_factor</span>
<span id="cb166-4"><a href="differential-gene-expression.html#cb166-4" tabindex="-1"></a></span>
<span id="cb166-5"><a href="differential-gene-expression.html#cb166-5" tabindex="-1"></a>geneSymbols <span class="ot">&lt;-</span> <span class="fu">mapIds</span>(org.Hs.eg.db, <span class="at">keys=</span><span class="fu">rownames</span>(count_values_up_and_down), <span class="at">column=</span><span class="st">&quot;SYMBOL&quot;</span>, <span class="at">keytype=</span><span class="st">&quot;ENTREZID&quot;</span>, <span class="at">multiVals=</span><span class="st">&quot;first&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="differential-gene-expression.html#cb168-1" tabindex="-1"></a><span class="fu">rownames</span>(count_values_up_and_down) <span class="ot">&lt;-</span> geneSymbols </span>
<span id="cb168-2"><a href="differential-gene-expression.html#cb168-2" tabindex="-1"></a></span>
<span id="cb168-3"><a href="differential-gene-expression.html#cb168-3" tabindex="-1"></a><span class="fu">pheatmap</span>(count_values_up_and_down, <span class="at">scale =</span> <span class="st">&quot;row&quot;</span>, <span class="at">show_rownames =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-107-1.png" width="672" /></p>
<h3>
Mapping with Ensemble DB
</h3>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="differential-gene-expression.html#cb169-1" tabindex="-1"></a>map_ids <span class="ot">&lt;-</span> <span class="cf">function</span>(ids)</span>
<span id="cb169-2"><a href="differential-gene-expression.html#cb169-2" tabindex="-1"></a></span>
<span id="cb169-3"><a href="differential-gene-expression.html#cb169-3" tabindex="-1"></a>    {</span>
<span id="cb169-4"><a href="differential-gene-expression.html#cb169-4" tabindex="-1"></a></span>
<span id="cb169-5"><a href="differential-gene-expression.html#cb169-5" tabindex="-1"></a>    symbol <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb169-6"><a href="differential-gene-expression.html#cb169-6" tabindex="-1"></a>    uniprot <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb169-7"><a href="differential-gene-expression.html#cb169-7" tabindex="-1"></a>    ensemble<span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb169-8"><a href="differential-gene-expression.html#cb169-8" tabindex="-1"></a>    gen_ids <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb169-9"><a href="differential-gene-expression.html#cb169-9" tabindex="-1"></a>    </span>
<span id="cb169-10"><a href="differential-gene-expression.html#cb169-10" tabindex="-1"></a>        <span class="cf">for</span> (gen <span class="cf">in</span> ids) {</span>
<span id="cb169-11"><a href="differential-gene-expression.html#cb169-11" tabindex="-1"></a>    </span>
<span id="cb169-12"><a href="differential-gene-expression.html#cb169-12" tabindex="-1"></a>   geneSymbol <span class="ot">&lt;-</span> <span class="fu">mapIds</span>(org.Hs.eg.db, <span class="at">keys=</span>gen, <span class="at">column=</span><span class="st">&quot;SYMBOL&quot;</span>, <span class="at">keytype=</span><span class="st">&quot;ENTREZID&quot;</span>, <span class="at">multiVals=</span><span class="st">&quot;first&quot;</span>)</span>
<span id="cb169-13"><a href="differential-gene-expression.html#cb169-13" tabindex="-1"></a></span>
<span id="cb169-14"><a href="differential-gene-expression.html#cb169-14" tabindex="-1"></a>   symbols <span class="ot">&lt;-</span> <span class="fu">unname</span>(geneSymbol) </span>
<span id="cb169-15"><a href="differential-gene-expression.html#cb169-15" tabindex="-1"></a>   ensembles <span class="ot">&lt;-</span> <span class="fu">mapIds</span>(org.Hs.eg.db, <span class="at">keys=</span>geneSymbol, <span class="at">column=</span><span class="st">&quot;ENSEMBL&quot;</span>, <span class="at">keytype=</span><span class="st">&quot;SYMBOL&quot;</span>, <span class="at">multiVals=</span><span class="st">&quot;first&quot;</span>)</span>
<span id="cb169-16"><a href="differential-gene-expression.html#cb169-16" tabindex="-1"></a>   ensembles <span class="ot">&lt;-</span> <span class="fu">unname</span>(ensembles )</span>
<span id="cb169-17"><a href="differential-gene-expression.html#cb169-17" tabindex="-1"></a>   uniprots<span class="ot">&lt;-</span><span class="fu">mapIds</span>(org.Hs.eg.db, <span class="at">keys=</span>geneSymbol, <span class="at">column=</span><span class="st">&quot;UNIPROT&quot;</span>, <span class="at">keytype=</span><span class="st">&quot;SYMBOL&quot;</span>, <span class="at">multiVals=</span><span class="st">&quot;first&quot;</span>)   </span>
<span id="cb169-18"><a href="differential-gene-expression.html#cb169-18" tabindex="-1"></a>   uniprots <span class="ot">&lt;-</span> <span class="fu">unname</span>(uniprots) </span>
<span id="cb169-19"><a href="differential-gene-expression.html#cb169-19" tabindex="-1"></a>   </span>
<span id="cb169-20"><a href="differential-gene-expression.html#cb169-20" tabindex="-1"></a>   ensemble[[<span class="fu">length</span>(ensemble)<span class="sc">+</span><span class="dv">1</span>]] <span class="ot">=</span> ensembles</span>
<span id="cb169-21"><a href="differential-gene-expression.html#cb169-21" tabindex="-1"></a>   gen_ids[[<span class="fu">length</span>(gen_ids)<span class="sc">+</span><span class="dv">1</span>]] <span class="ot">=</span> gen</span>
<span id="cb169-22"><a href="differential-gene-expression.html#cb169-22" tabindex="-1"></a>   symbol[[<span class="fu">length</span>(symbol)<span class="sc">+</span><span class="dv">1</span>]]<span class="ot">=</span>symbols</span>
<span id="cb169-23"><a href="differential-gene-expression.html#cb169-23" tabindex="-1"></a>   uniprot[[<span class="fu">length</span>(uniprot)<span class="sc">+</span><span class="dv">1</span>]]<span class="ot">=</span>uniprots</span>
<span id="cb169-24"><a href="differential-gene-expression.html#cb169-24" tabindex="-1"></a>        }</span>
<span id="cb169-25"><a href="differential-gene-expression.html#cb169-25" tabindex="-1"></a>    </span>
<span id="cb169-26"><a href="differential-gene-expression.html#cb169-26" tabindex="-1"></a>    df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">entrez=</span><span class="fu">unlist</span>(gen_ids), <span class="at">symbol=</span><span class="fu">unlist</span>(symbol),<span class="at">ensemble=</span><span class="fu">unlist</span>(ensemble), <span class="at">uniport=</span><span class="fu">unlist</span>(uniprot) ) <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>()</span>
<span id="cb169-27"><a href="differential-gene-expression.html#cb169-27" tabindex="-1"></a>  <span class="co">#print(df)</span></span>
<span id="cb169-28"><a href="differential-gene-expression.html#cb169-28" tabindex="-1"></a>    <span class="fu">return</span>(df)</span>
<span id="cb169-29"><a href="differential-gene-expression.html#cb169-29" tabindex="-1"></a>    </span>
<span id="cb169-30"><a href="differential-gene-expression.html#cb169-30" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="differential-gene-expression.html#cb170-1" tabindex="-1"></a>ids <span class="ot">&lt;-</span>  <span class="fu">c</span>(<span class="st">&quot;3175&quot;</span>,<span class="st">&quot;10202&quot;</span>, <span class="st">&quot;388228&quot;</span> )</span>
<span id="cb170-2"><a href="differential-gene-expression.html#cb170-2" tabindex="-1"></a><span class="fu">map_ids</span>(ids)</span></code></pre></div>
<pre><code>## &#39;select()&#39; returned 1:1 mapping between keys and columns
## &#39;select()&#39; returned 1:1 mapping between keys and columns
## &#39;select()&#39; returned 1:1 mapping between keys and columns
## &#39;select()&#39; returned 1:1 mapping between keys and columns
## &#39;select()&#39; returned 1:1 mapping between keys and columns
## &#39;select()&#39; returned 1:1 mapping between keys and columns
## &#39;select()&#39; returned 1:1 mapping between keys and columns
## &#39;select()&#39; returned 1:1 mapping between keys and columns
## &#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>
<pre><code>## # A tibble: 3 × 4
##   entrez symbol  ensemble        uniport
##   &lt;chr&gt;  &lt;chr&gt;   &lt;chr&gt;           &lt;chr&gt;  
## 1 3175   ONECUT1 ENSG00000169856 Q9UBC0 
## 2 10202  DHRS2   ENSG00000100867 Q13268 
## 3 388228 SBK1    ENSG00000188322 Q52WX2</code></pre>
<h3>
Go stats analyse
</h3>
<h3>
Function calling for upregulated genes
</h3>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="differential-gene-expression.html#cb173-1" tabindex="-1"></a><span class="fu">go_analyes_function</span> (dge_results_one, <span class="dv">1</span>)</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-111-1.png" width="672" /></p>
<h3>
Function calling for downregulated genes
</h3>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="differential-gene-expression.html#cb174-1" tabindex="-1"></a><span class="fu">go_analyes_function</span> (dge_results_one, <span class="dv">0</span>)</span></code></pre></div>
<p><img src="bookdownproj_files/figure-html/unnamed-chunk-112-1.png" width="672" /></p>
<h3>
Conclusion
</h3>
<p>Treatment with ONECUT induces reprogramming of fibroblasts into neurons and expression of neuronal genes,
regulates DNA transcription, synthesis of RNA and transcription by RNA polymerase II.</p>
<p>ONECUT1 and BCLXL have antagonistic activity.This can be seen from the heatmaps and PCA map.
The treatment with ONECUT1 activates genes : 3175, 10202, 388228, 338440, 284422</p>
<ul>
<li>ONECUT1. This gene gene encodes a member of the ONECUT family of transcription factors,
induced neuronal morphology and expression of neuronal genes. (Gene summary Entrez,<a href="https://www.proteinatlas.org/ENSG00000169856-ONECUT1" class="uri">https://www.proteinatlas.org/ENSG00000169856-ONECUT1</a>).</li>
<li>DHRS2. This gene encodes enzyme that activates hormones, such as steroid hormones, prostaglandins;
reduces proliferation, migration and invasion of cancer cells.
(Gene summary Entrez, <a href="https://www.proteinatlas.org/ENSG00000100867-DHRS2" class="uri">https://www.proteinatlas.org/ENSG00000100867-DHRS2</a>)</li>
<li>SBK1. This gene enables protein serine/threonine kinase activity (Gene summary Entrez,<a href="https://www.proteinatlas.org/ENSG00000188322-SBK1" class="uri">https://www.proteinatlas.org/ENSG00000188322-SBK1</a>)</li>
<li>ANO9. The protein encoded by this gene is a member of the TMEM16 family of proteins, some of which form integral membrane calcium-activated chloride channels.(Gene summary Entrez,<a href="https://www.proteinatlas.org/ENSG00000185101-ANO9" class="uri">https://www.proteinatlas.org/ENSG00000185101-ANO9</a>)</li>
<li>SMIM24 This gene encodes the protein predicted to be located in membrane. Predicted to be integral component of membrane.(Gene summary Entrez,<a href="https://www.proteinatlas.org/ENSG00000095932-SMIM24" class="uri">https://www.proteinatlas.org/ENSG00000095932-SMIM24</a>)</li>
</ul>
<p></br>
From the Go-terms enrichment analysis we can conclude that treatment with ONECUT suppresses gene activity
associated with BCLXL: antiapoptotic genes, proliferation, cancer cell migration.</p>

</div>
<div id="footer">Copyright &copy;  Hogeschool Utrecht   DSFB2  <i class="fa-solid fa-webhook"></i> 2024  		</footer>




            </section>

          </div>
        </div>
      </div>
<a href="alignment.-ruwe-gegevens-inspecteren.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="creating-r-packages.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
